src/Components/Panels/Toolbox/BaseControl.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+
+import { IBase } from '../../../Document/Obstacles/Base';
+
+import styles from './BaseControl.module.scss';
+
+interface Props {
+  data: IBase;
+  onChange: (changes: IBase) => void;
+}
+
+const BaseControl = ({ data, onChange }: Props) => {
+  const handleBaseColourOnChange = (value: number) => {
+    onChange({
+      ...data,
+      color: value as IBase['color'],
+    });
+  };
```

Comment:
```suggestion
  const handleBaseColorOnChange = (event: ChangeEvent<HTMLSelectElement>) => {
    onChange({
      ...data,
      color: +event.currentTarget.value as IBase['color'],
    });
  };
```

- [`currentTarget` vs `target` (simple explanation)](https://stackoverflow.com/questions/10086427/what-is-the-exact-difference-between-currenttarget-property-and-target-property) and [an example of when it'd matter](https://kbarker.dev/blog/the-difference-between-event-target-and-event-current-target/); usually we want to listen on an event that the listener was attached to and not another element up the DOM tree that could have triggered this through [event bubbling](https://www.sitepoint.com/event-bubbling-javascript/)
- [parseInt "short-hand"](https://stackoverflow.com/a/6683020); not actually a short-hand for `parseInt` but it performs a [similar enough functionality](https://stackoverflow.com/q/17106681) in this case.